decomp/semantics/uds.py

```diff
@@ -323,25 +413,59 @@ def semantic_type_subspaces(self) -> Dict[str, Set[str]]:
         return dict(self.semantic_node_type_subspaces,
                     **self.semantic_edge_type_subspaces)
 
+class UDSGraph(ABC):
+    """Abstract base class for sentence- and document-level graphs"""
```

Comment:
If you're on board with abstracting out a base class for both graph types like this, let me know what other methods you think should be moved into it. (I was planning on at least adding `add_annotation`, too—I just didn't get to that yet.)

Reply:
I think this is a good idea.